org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/SonarLintRuleBrowser.java

```diff
@@ -35,51 +36,59 @@
 import org.sonarlint.eclipse.core.internal.utils.StringUtils;
 import org.sonarlint.eclipse.ui.internal.SonarLintImages;
 import org.sonarlint.eclipse.ui.internal.properties.RulesConfigurationPage;
-import org.sonarsource.sonarlint.core.client.api.common.RuleDetails;
-import org.sonarsource.sonarlint.core.client.api.connected.ConnectedRuleDetails;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneRuleDetails;
 import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneRuleParam;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetEffectiveRuleDetailsResponse;
+import org.sonarsource.sonarlint.core.clientapi.backend.rules.GetStandaloneRuleDescriptionResponse;
 
+/** "SonarLint Rule Description" view content (monolithic / educational rule description) */
 public class SonarLintRuleBrowser extends SonarLintWebView {
-
-  private RuleDetails ruleDetails;
+  private Either<GetEffectiveRuleDetailsResponse, GetStandaloneRuleDescriptionResponse> ruleDetailsResponse;
```

Comment:
I would try to avoid leaking the "Response" objects everywhere. This might be done later in other subtasks, but I think this `SonarLintRuleBrowser` component should only deal with `Either<RuleMonolithicDescriptionDto, RuleSplitDescriptionDto> description`